modules/siddhi-extensions/input-transports/jms-input-transport/src/main/java/org/wso2/siddhi/extension/input/transport/jms/JMSInputTransport.java

```diff
@@ -0,0 +1,217 @@
+    private OptionHolder optionHolder;
+    private JMSConnectionFactory jmsConnectionFactory;
+    private JMSListener jmsListener;
+    private ExecutionPlanContext executionPlanContext;
+
+    private int minThreadPoolSize;
+    private int maxThreadPoolSize;
+    private int jobQueueSize;
+    private String destination;
+    private int KeepAliveTimeInMillis;
+    private String inputAdapterName;
+
+    @Override
+    public void init(SourceEventListener sourceEventListener, OptionHolder optionHolder,
+                     ExecutionPlanContext executionPlanContext) {
+        this.sourceEventListener = sourceEventListener;
+        this.optionHolder = optionHolder;
+        this.executionPlanContext = executionPlanContext;
+
+        minThreadPoolSize = JMSInputTransportConstants.DEFAULT_MIN_THREAD_POOL_SIZE;
+        maxThreadPoolSize = JMSInputTransportConstants.DEFAULT_MAX__THREAD_POOL_SIZE;
+        //todo: get these from a config file
+        jobQueueSize = JMSInputTransportConstants.DEFAULT_JOB_IN_QUEUE_SIZE;
+        KeepAliveTimeInMillis = JMSInputTransportConstants.DEFAULT_KEEP_ALIVE_TIME_IN_INTERVAL;
+        inputAdapterName = "JMS-INPUT-" + sourceEventListener.getStreamDefinition().getId();
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        Map<String, String> jmsConnectionProperties = initJMSProperties();
+        destination = optionHolder.validateAndGetStaticValue(JMSInputTransportConstants.ADAPTER_JMS_DESTINATION);
+        //creating the connection factory
+        jmsConnectionFactory = new JMSConnectionFactory(convertMapToHashTable(jmsConnectionProperties),
+                inputAdapterName);
+        Map<String, String> messageConfig = new HashMap<>();
+        messageConfig.put(JMSInputTransportConstants.ADAPTER_JMS_DESTINATION, destination);
+        JMSTaskManager jmsTaskManager = JMSTaskManagerFactory.createTaskManagerForService(jmsConnectionFactory,
+                inputAdapterName, new NativeWorkerPool(minThreadPoolSize, maxThreadPoolSize, KeepAliveTimeInMillis,
+                        jobQueueSize, "JMS Threads", "JMSThreads" + UUID.randomUUID().toString()), messageConfig);
+        jmsTaskManager.setJmsMessageListener(new JMSMessageListener(sourceEventListener));
+
+        jmsListener = new JMSListener(inputAdapterName + "#" + destination, jmsTaskManager);
+        jmsListener.startListener();    }
+
+    @Override
+    public void disconnect() {
+        if (jmsListener != null) {
+            jmsListener.stopListener();
+        }
+
+        if (jmsConnectionFactory != null) {
+            jmsConnectionFactory.stop();
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("JMS consumer " + inputAdapterName + " disconnected from destination : " + destination);
+        }
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public void pause() {
+
+    }
+
+    @Override
+    public void resume() {
+
+    }
```

Comment:
Can you add TODOs for these unimplemented methods assuming you gonna implement them?